File: 3270-find-the-key-of-the-numbers/3270-find-the-key-of-the-numbers.java
class Solution {
    public int generateKey(int num1, int num2, int num3) {
        String s1 = augment(num1);
        String s2 = augment(num2);
        String s3 = augment(num3);
        
        String [] arr = new String [] {s1, s2, s3};
        
        String res = "";
        
        for (int i = 0; i < 4; ++i) {
            int result = 10;
            int val;
            for (String s : arr) {
                val = s.charAt(i) - '0';
                result = Math.min(val, result);
            }
            
            res += result;
        }
        
        return Integer.parseInt(res);
    }
    
    private String augment(int num1) {
        String s1 = num1 + "";
        
        while (s1.length() < 4) {
            s1 = "0" + s1;
        }
        
        return s1;
    }
}

File: 3270-find-the-key-of-the-numbers/3270-find-the-key-of-the-numbers.cpp
class Solution {
public:
    int generateKey(int num1, int num2, int num3) {

      auto to_str = [&](int x) {
        string s = to_string(x);
        while (s.size() < 4) {
          s = "0" + s;
        }
        return s;
      };

      auto n1 = to_str(num1);
      auto n2 = to_str(num2);
      auto n3 = to_str(num3);

      int ret = 0;
      for (int i = 0; i < 4; ++i) {
        int x = min(n1[i], min(n2[i], n3[i])) - '0';
        ret = ret * 10 + x;
      }
      return ret;
    }
};